discovery/uyuni/uyuni.go

```diff
@@ -308,25 +307,29 @@ func (d *Discovery) getTargetsForSystems(
 	return result, nil
 }
 
-func (d *Discovery) refresh(ctx context.Context) ([]*targetgroup.Group, error) {
+func (d *Discovery) refresh(_ context.Context) ([]*targetgroup.Group, error) {
 	rpcClient, err := xmlrpc.NewClient(d.apiURL.String(), d.roundTripper)
 	if err != nil {
 		return nil, err
 	}
 	defer rpcClient.Close()
 
-	token, err := login(rpcClient, d.username, d.password)
-	if err != nil {
-		return nil, errors.Wrap(err, "unable to login to Uyuni API")
-	}
-	defer func() {
-		if err := logout(rpcClient, token); err != nil {
-			level.Debug(d.logger).Log("msg", "Failed to log out from Uyuni API", "err", err)
+	// If the current token has expired, refresh it
+	if time.Now().After(d.tokenExpiration) {
+		// Uyuni API doesn't state the unit of measure for the duration.
+		// From looking at the code, I believe it is in seconds.
+		d.token, err = login(rpcClient, d.username, d.password, int(tokenDuration.Seconds()))
+		if err != nil {
+			return nil, errors.Wrap(err, "unable to login to Uyuni API")
 		}
-	}()
+		// login at half the token lifetime
+		d.tokenExpiration = time.Now().Add(tokenDuration / 2)
```

Comment:
If we set `tokenDuration` to half of its value and repeat for maximum 10 times we'll be on the safe side.
Otherwise we should make it configurable.

Reply:
Currently `tokenDuration` is 10 minutes, so it will refresh every 5 min. I'm not sure how often it's refreshed, so I don't know how many times the token will get used. Is there a limit on the number times the token can be used?

If the consensus is the duration should be added to the config, I can do that.

Reply:
> If the consensus is the duration should be added to the config, I can do that.

Seems to be a simple and sensible approach to me. @roidelapluie, what do you think?

Reply:
I went ahead and moved the login interval/duration to an optional config item with the same default of 10 minutes. I also updated the configuration documentation. This was my first config change, so let me know if I missed something.

Reply:
I don't think we should expose users to that option. I'd rather have a default of 10 minutes not configurable, and do not even mention it in the documentation. This should be an implementation detail.

Reply:
Ok, I'll revert that change.